compilers/concrete-optimizer/concrete-optimizer/src/optimization/dag/multi_parameters/analyze.rs

```diff
@@ -207,14 +206,13 @@ impl VariancedDag {
                     nb_partitions,
                     operator.partition().instruction_partition,
                 ),
-                Operator::LevelledOp { manp, .. } => {
-                    let max_var = operator
-                        .get_inputs_iter()
-                        .map(|a| a.variance()[operator.partition().instruction_partition].clone())
-                        .reduce(max_var)
-                        .unwrap();
-                    max_var.after_levelled_op(*manp)
-                }
+                Operator::LevelledOp { weights, .. } => operator
+                    .get_inputs_iter()
+                    .zip(weights)
+                    .fold(SymbolicVariance::ZERO, |acc, (inp, weight)| {
+                        acc + inp.variance()[operator.partition().instruction_partition].clone()
+                            * (*weight * *weight)
```

Comment:
use square function to avoid the * confusion (see Dot case) (or remove the extra *)

Reply:
also iterating over weights instead to the trick, see Dot case

Reply:
Done thanks !